thunder/core/rematerialization.py

```diff
@@ -87,7 +87,9 @@ def find_external_consumer_inputs(
         Tuple[ProxyInterface, ...]: Consumer's inputs that must be included in
         the input of the consumer.
     """
-    all_produced_vars = tuple(chain.from_iterable((y for y in x.flat_outs) for x in producer.subsymbols))
+    all_produced_vars = tuple(
+        chain.from_iterable((y for y in x.flat_outs if y is not None) for x in producer.subsymbols)
```

Comment:
Here's a better alternative because further code assumes we're iterating over proxies accessing its name.
```suggestion
        chain.from_iterable((y for y in x.flat_proxy_outs) for x in producer.subsymbols)
```